runtime/Java/src/org/antlr/v4/runtime/Parser.java

```diff
@@ -249,6 +249,15 @@ public Token matchWildcard() throws RecognitionException {
 		return t;
 	}
 
+	protected ParserRuleContext createContext(int ruleIndex, int altIndex, ParserRuleContext parent, int invokingState, Object... arguments) {
```

Comment:
💭 Right now `arguments` is available but not used. Options include:

1. Omit `arguments` and rely on the context constructor that doesn't take any
2. Require `arguments` match the form of the context constructor that includes all arguments
3. Allow `arguments` to be omitted by the called (use the context constructor that doesn't include them), or specified (use the context constructor that passes arguments)

If we do allow arguments, we will likely need a helper function like this:

```
private static Object getArgumentOrDefault(Object[] arguments, int index, Object defaultValue) {
  if (arguments == null || index >= arguments.length || arguments[index] == null) {
    return defaultValue;
  }

  return arguments[index];
}
```

Reply:
➡️ Implemented the third option (arguments are optional)